skimage/filters/_unsharp_mask.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import division
+import numpy as np
+from scipy.ndimage.filters import gaussian_filter
+from skimage.util.dtype import dtype_limits
+from skimage import img_as_float
+
+
+def _unsharp_mask_single_channel(image, radius, amount, vrange):
+    """Single channel implementation of the unsharp masking filter."""
+
+    blurred = gaussian_filter(image,
+                              sigma=radius,
+                              mode='reflect')
+
+    result = image + (image - blurred) * amount
+    if vrange is not None:
+        return np.clip(image, vrange[0], vrange[1], out=result)
+    return result
+
+
+def unsharp_mask(image, radius=1.0, amount=1.0, multichannel=False, preserve_range=False):
+    """Unsharp masking filter.
+
+    The sharp details are identified as the difference between the original
+    image and its blurred version. These details are then scaled, and added
+    back to the original image.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input image.
+    radius : scalar or sequence of scalars, optional
+        If a scalar is given, then its value is used for all dimensions.
+        If sequence is given, then there must be exactly one radius
+        for each dimension except the last dimension for multichannel images.
+        Note that 0 radius means no blurring, and negative values are
+        not allowed.
+    amount : scalar, optional
+        The details will be amplified with this factor. The factor could be 0
+        or negative. Typically, it is a small positive number, e.g. 1.0.
+    multichannel : bool, optional
+        If True, the last `image` dimension is considered as a color channel,
+        otherwise as spatial. Color channels are processed individually.
+    preserve_range: bool, optional
+        If False, the integer/unsigned types will be scaled to [-1,1]/[0,1]
+        before applying unsharp masking filter. If the output exceeds this limit,
+        it will be clipped to [-1,1] / [0,1].
+        If True, then the original ranges will be kept in the input, but the
+        values will be converted to floats.
+        See more details at the data types section in the documentation.
+
+    Returns
+    -------
+    output : ndarray
+        Image with unsharp mask applied. It returns float type, regardless
+        of the input type. The user is responsible to cast the result to
+        the expected type. If the
```

Comment:
Unfinished sentence here.

Reply:
Indeed, however, I cannot remember what was meant there. let me know, if anything is felt missing from the documentation.

Reply:
I think it is fine "as is".